ait/core/server/plugin.py

```diff
@@ -0,0 +1,138 @@
+from collections import defaultdict
+import importlib
+import re
+
+import gevent
+import gevent.monkey; gevent.monkey.patch_all()
+
+import ait.core
+from ait.core import  db, limits, log, notify, tlm
+from client import ZMQInputClient
+
+
+class Plugin(ZMQInputClient):
+    """
+    This is the parent class for all plugins. All plugins must implement
+    their own process method which is called when a message is received.
+    """
+
+    def __init__(self, inputs, outputs, zmq_args={}, **kwargs):
+        self.type = 'Plugin'
```

Comment:
What are we doing with `type`? Do we need this?

Reply:
Update usage to just call type() on class, removing need for type attribute